src/main/java/de/hpi/bpt/argos/storage/PersistenceAdapterImpl.java

```diff
@@ -397,7 +399,31 @@ public int getEntityTypesCount() {
 				.setFirstResult(listStartIndex)
 				.setMaxResults(listEndIndex);
 
-		return databaseAccess.getArtifacts(session, query, transaction, query::getResultList, new ArrayList<>());
+		List<Event> events = databaseAccess.getArtifacts(session, query, transaction, query::getResultList, new ArrayList<>());
+
+		// we need to fetch more events, because there are still entityIds, which are not covered and we did not get enough events yet
+		if (events.size() < Math.abs(listEndIndex - listStartIndex) && limitedEntityIds.size() < entityIds.length) {
+			List<Long> entityIdsLeft = Arrays.asList(entityIds);
+			entityIdsLeft.removeAll(limitedEntityIds);
+
+			events.addAll(getEvents(eventTypeId, 0, Math.abs(listEndIndex - listStartIndex) - events.size(), entityIdsLeft.toArray(new Long[entityIdsLeft.size()])));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 156 characters long line (which is greater than 150 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS00103)